src/main/java/com/hida/DatabaseManager.java

```
@@ -0,0 +1,261 @@
+            String sqlQuery = String.format("SELECT %4$s FROM %s "
+                    + "WHERE %4$s LIKE '%2$s%3$s';", 
+                    TABLE_NAME, prefix, pad, COLUMN_NAME);
+
+            // retrieve results
+            ResultSet databaseResponse
+                    = databaseStatement.executeQuery(sqlQuery);
+
+            // returns the size of the results (number of ids). If nothing was
+            // found by the database, 0 is returned instead. 
+            if (databaseResponse.next()) {
+                return databaseResponse.getFetchSize();
+            } else {
+                return 0;
+            }
+
+        } catch (SQLException exception) {
+            System.err.println(exception.getMessage() + " in checkPrefix");
+            System.err.println(Arrays.toString(exception.getStackTrace()));
+        }
+        // error value, will revise later
+        return -1;
+
+    }
+
+    /**
+     * Prints a list of ids to the server. Strictly used for error checking.
+     */
+    private void printData() {
+        System.out.println("current list of items: ");
+        try {
+            Statement s = DATABASE_CONNECTION.createStatement();
+            String sql = String.format("SELECT * FROM %s;", TABLE_NAME);
+            ResultSet r = s.executeQuery(sql);
+ 
+            for (int i = 0; r.next(); i++) {
+                String curr = r.getString("id");
+                System.out.printf("id(%d):  %s", i, curr);
+            }
+            System.out.println("done");
+        } catch (SQLException e) {
+            System.err.println("printData: " + e.getMessage());
+        }
+
+    }
+
+    /**
+     * Used by an external method to close this connection. 
+     */
+    public synchronized void closeConnection() {
+        try {
+            DATABASE_CONNECTION.close();
+        } catch (SQLException exception) {
+            System.out.println(exception.getMessage());

```

See comment about ignoring exceptions